src/Features/CSharp/Portable/Completion/CompletionProviders/AwaitCompletionProvider.cs

```diff
@@ -0,0 +1,139 @@
+
+            if (!syntaxContext.IsAwaitKeywordContext(position))
+            {
+                return;
+            }
+
+            var method = syntaxContext.TargetToken.GetAncestor(node => node.IsAsyncSupportingFunctionSyntax());
+            var shouldMakeContainerAsync = method is not null && !method.GetModifiers().Any(SyntaxKind.AsyncKeyword);
+            context.AddItem(GetCompletionItem(shouldMakeContainerAsync));
+
+            return;
+
+            static CompletionItem GetCompletionItem(bool shouldMakeContainerAsync)
+                => CommonCompletionItem.Create(
+                       displayText: SyntaxFacts.GetText(SyntaxKind.AwaitKeyword),
+                       displayTextSuffix: "",
+                       rules: CompletionItemRules.Default,
+                       Glyph.Keyword,
+                       description: RecommendedKeyword.CreateDisplayParts(SyntaxFacts.GetText(SyntaxKind.AwaitKeyword), string.Empty),
+                       inlineDescription: shouldMakeContainerAsync ? CSharpFeaturesResources.Make_container_async : null,
+                       isComplexTextEdit: shouldMakeContainerAsync);
+        }
+
+        public override async Task<CompletionChange> GetChangeAsync(Document document, CompletionItem item, char? commitKey = null, CancellationToken cancellationToken = default)
+        {
+            // IsComplexTextEdit is true when we want to add async to the container.
+            if (!item.IsComplexTextEdit)
+            {
+                return await base.GetChangeAsync(document, item, commitKey, cancellationToken).ConfigureAwait(false);
+            }
+
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var declaration = root.FindToken(item.Span.Start).GetAncestor(node => node.IsAsyncSupportingFunctionSyntax());
+            if (declaration is null)
+            {
+                // We already check that in ProvideCompletionsAsync above.
+                Debug.Assert(false, "Expected non-null value for declaration.");
+                return await base.GetChangeAsync(document, item, commitKey, cancellationToken).ConfigureAwait(false);
+            }
+
+            var documentWithAsyncModifier = document.WithSyntaxRoot(root.ReplaceNode(declaration, AddAsyncModifier(declaration)));
+            using var _ = ArrayBuilder<TextChange>.GetInstance(out var builder);
+
+            builder.AddRange(await documentWithAsyncModifier.GetTextChangesAsync(document, cancellationToken).ConfigureAwait(false));
+            builder.Add(new TextChange(item.Span, item.DisplayText));
+
+            var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            var newText = text.WithChanges(builder);
+            return CompletionChange.Create(CodeAnalysis.Completion.Utilities.Collapse(newText, builder.ToImmutableArray()));
+        }
+
+        private static SyntaxNode AddAsyncModifier(SyntaxNode declaration)
```

Comment:
since we're manually adding the async modifier anyways, it's fairly trivial for us to instead just determine the text-edit for where the async modifier should go directly.  please switch to that.  thanks!